File: packages/iroh-native/ios/HappierIrohNativeModule.swift
import ExpoModulesCore
import Foundation

/// Lifecycle/status only; tunnel bytes never cross this boundary.
public final class HappierIrohNativeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("HappierIrohNative")
    Function("getAvailability") { () -> [String: Any] in
      ["available": false, "platform": "ios", "engine": "iroh", "supportsHomeTunnel": false]
    }
    AsyncFunction("startHomeTunnel") { (_ request: [String: Any]) async throws -> [String: Any] in
      throw NSError(domain: "HappierIrohNative", code: 1, userInfo: [NSLocalizedDescriptionKey: "Iroh native engine is not linked in this build."])
    }
    AsyncFunction("stopHomeTunnel") { (_ leaseId: String) async throws -> Void in }
    AsyncFunction("getHomeTunnelStatus") { (_ homeServerIdentityId: String) async -> [String: Any]? in nil }
  }
}

File: packages/iroh-native/android/src/main/java/dev/happier/iroh/HappierIrohNativeModule.kt
package dev.happier.iroh

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/** Lifecycle/status only; native Rust owns all stream bytes. */
class HappierIrohNativeModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("HappierIrohNative")
    Function("getAvailability") { mapOf("available" to false, "platform" to "android", "engine" to "iroh", "supportsHomeTunnel" to false) }
    AsyncFunction("startHomeTunnel") { _: Map<String, Any?> -> throw IllegalStateException("Iroh native engine is not linked in this build.") }
    AsyncFunction("stopHomeTunnel") { _: String -> Unit }
    AsyncFunction("getHomeTunnelStatus") { _: String -> null }
  }
}
